exec/plugin-commands-script-runners/src/exec.ts

```diff
@@ -183,34 +205,48 @@ export async function handler (
             stdio: 'inherit',
             shell: opts.shellMode ?? false,
           })
-          result.passes++
+          result[prefix].status = 'passed'
+          result[prefix].duration = getExecutionDuration(startTime)
         } catch (err: any) { // eslint-disable-line
           if (!opts.recursive && typeof err.exitCode === 'number') {
             exitCode = err.exitCode
             return
           }
           logger.info(err)
 
+          result[prefix] = {
+            status: 'failure',
+            duration: getExecutionDuration(startTime),
+            error: err,
+            message: err.message,
+            prefix,
+          }
+
           if (!opts.bail) {
-            result.fails.push({
-              error: err,
-              message: err.message,
-              prefix,
-            })
             return
           }
 
           if (!err['code']?.startsWith('ERR_PNPM_')) {
             err['code'] = 'ERR_PNPM_RECURSIVE_EXEC_FIRST_FAIL'
           }
           err['prefix'] = prefix
+          await writeRecursiveSummary({
+            dir: opts.lockfileDir ?? opts.dir,
+            summary: result,
+            reportSummary: opts.reportSummary,
+          })
           /* eslint-enable @typescript-eslint/dot-notation */
           throw err
         }
       }
       )))
   }
 
+  await writeRecursiveSummary({
+    dir: opts.lockfileDir ?? opts.dir,
+    summary: result,
+    reportSummary: opts.reportSummary,
```

Comment:
why do you need to call the function if reportSummary is false? Just add an if statement

Reply:
Thanks for you adive, will update it now~